meta/store.go

```diff
@@ -843,6 +846,19 @@ func (s *Store) ShardGroupByTimestamp(database, policy string, timestamp time.Ti
 	return
 }
 
+func (s *Store) ShardOwner(shardID uint64) (database, policy string, sgi *ShardGroupInfo) {
```

Comment:
Yeah, I think this should not return deleted shards. I think we need to pretty clear that deleted shards are invisible to the rest of the system (except for one call, `DeletedShards`, on the retention policy type).